src/coreclr/vm/interpexec.cpp

```diff
@@ -83,6 +88,108 @@ static void InterpBreakpoint()
+{
+    size_t helperDirectOrIndirect = (size_t)dataItem;
+    if (helperDirectOrIndirect & INTERP_INDIRECT_HELPER_TAG)
+        return *(THelper *)(helperDirectOrIndirect & ~INTERP_INDIRECT_HELPER_TAG);
+    else
+        return (THelper)helperDirectOrIndirect;
+}
+
+template <typename TResult, typename TSource> static void ConvFpHelper(int8_t *stack, const int32_t *ip)
+{
+    static_assert(!std::numeric_limits<TSource>::is_integer, "ConvFpHelper is only for use on floats and doubles");
+    static_assert(std::numeric_limits<TResult>::is_integer, "ConvFpHelper is only for use on floats and doubles to be converted to integers");
+
+    // First, promote the source value to double
+    double src = LOCAL_VAR(ip[2], TSource),
+        minValue = (double)std::numeric_limits<TResult>::lowest(),
+        maxValue = (double)std::numeric_limits<TResult>::max();
+
+    // (src != src) checks for NaN, then we check whether the min and max values (as represented by their closest double)
+    //  properly bound the source value so that when it is truncated it will be in range
+    // We assume that we are in round-towards-zero mode. For NaN we want to return 0, and for out of range values, saturate.
+    TResult result;
+    if (src != src)
+        result = 0;
+    else if (src >= maxValue)
+        result = std::numeric_limits<TResult>::max();
+    else if (std::numeric_limits<TResult>::is_signed && (src <= -1))
+        result = 0;
+    else if (src < minValue)
+        result = std::numeric_limits<TResult>::lowest();
+    else
+        result = (TResult)src;
+
+    // According to spec, for result types smaller than int32, we store them on the stack as int32
+    if (sizeof(TResult) >= 4)
+        LOCAL_VAR(ip[1], TResult) = result;
+    else
+        LOCAL_VAR(ip[1], int32_t) = (int32_t)result;
+}
+
+template <typename TResult, typename TSource> static void ConvOvfFpHelper(int8_t *stack, const int32_t *ip)
+{
+    static_assert(!std::numeric_limits<TSource>::is_integer, "ConvOvfFpHelper is only for use on floats and doubles");
+    static_assert(std::numeric_limits<TResult>::is_integer, "ConvOvfFpHelper is only for use on floats and doubles to be converted to integers");
+
+    // First, promote the source value to double
+    double src = LOCAL_VAR(ip[2], TSource),
+        minValue = (double)std::numeric_limits<TResult>::lowest(),
+        maxValue = (double)std::numeric_limits<TResult>::max();
+
+    bool outOfRange;
+    // (src != src) checks for NaN, then we check whether the min and max values (as represented by their closest double)
+    //  properly bound the source value so that when it is truncated it will be in range
+    // We assume that we are in round-towards-zero mode
+    if (std::numeric_limits<TResult>::is_signed)
+        outOfRange = (src != src) || (src < minValue) || (src >= maxValue);
+    else
+        outOfRange = (src != src) || (src <= -1) || (src >= maxValue);
```

Comment:
```suggestion
        outOfRange = (src != src) || (src <= -1) || (src > maxValue);
```
I think it should be `>` for unsigned types given how to values round up and down. (You can do a quick test for the last convertible value and first overflowing value to verify.)

Alternatively, you can copy to constants and exact logic from the managed helpers as discussed in the other thread.